Update driver changelogs for uploads feature and multimethods (#31685)
metabase/metabase: docs/developers-guide/driver-changelog.md

## Metabase 0.47.0

- A new driver feature has been added: `:schemas`. This feature signals whether the database organizes tables in
  schemas (also known as namespaces) or not. Most databases have schemas so this feature is on by default.
  An implemention of the multimethod `metabase.driver/database-supports?` for `:schemas` is required only if the
  database doesn't store tables in schemas.

- Another driver feature has been added: `uploads`. The `uploads` feature signals whether the database supports 
  uploading CSV files to tables in the database. To support the uploads feature, implement the following new 
  multimethods: `metabase.driver/create-table!` (creates a table), `metabase.driver/drop-table!` (drops
  a table), and `metabase.driver/insert-into!` (inserts values into a table).

- The multimethod `metabase.driver/syncable-schemas` has been added. This method is used to list schemas to upload

>   CSVs to, and it should include all schemas that are able to be synced to. Currently it only needs to be implemented

Reviewer: I'm not a native English speaker so feel free to ignore.
```suggestion
  CSVs to, and it should include all schemas that can be synced. Currently it only needs to be implemented
```

Reviewer 2: I am, and I agree this edit is clearer.

>   if the database has schema, and the database supports the `uploads` feature.

Reviewer: ```suggestion
  if the database has schema, and the database supports the `:uploads` feature.
```
